File: test/fixtures/java-unirest/read_cookies_and_mixed_auth.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.get("http://localhost:28139")
            .basicAuth("user", "pass")
            .header("Authorization", "Bearer AAAAAAAAAAAA")
            .asString();
    }
}

File: test/fixtures/kotlin-ktor/read_cookies_and_mixed_auth.kt
import io.ktor.client.HttpClient
import io.ktor.client.plugins.auth.Auth
import io.ktor.client.plugins.auth.providers.BasicAuthCredentials
import io.ktor.client.plugins.auth.providers.basic
import io.ktor.client.request.get
import io.ktor.client.request.header
import io.ktor.client.statement.bodyAsText
import io.ktor.http.isSuccess
import kotlinx.coroutines.runBlocking

val client = HttpClient{
    // requires the io.ktor:ktor-client-auth dependency
    install(Auth) {
        basic {
            credentials {
                BasicAuthCredentials("user", "pass")
            }
            realm = "Access to the '/' path"        }
    }
}

runBlocking {
    val response = client.get("http://localhost:28139") {
        header("Authorization", "Bearer AAAAAAAAAAAA")
    }
    if (!response.status.isSuccess()) throw Exception("Request failed with status code ${response.status}")

    println(response.bodyAsText())
}
